kyaml/yaml/fns.go

```diff
@@ -612,6 +612,39 @@ func Set(value *RNode) FieldSetter {
 	return FieldSetter{Value: value}
 }
 
+// MapEntrySetter sets a field or map entry to a value.
+type MapEntrySetter struct {
+	// Name is the name of the field or key to lookup in a MappingNode.
+	// If Name is unspecified, it will use the Key's Value
+	Name string `yaml:"name,omitempty"`
```

Comment:
> What if Name is present but doesn't match Key's value? 

Well, in this case, it will replace the key with that name to the Key's value.

> Some of these filters support string options in addition to RNode (should be mutually exclusive). Might having StringKey and StringValue be more helpful than a lookup-only Key option?

I believe that if you want to set a string key/value the `FieldSetter` would do great

> Why did you need to create MapEntrySetter when FieldSetter handles maps? (Is it because you want to set the node for the key as well?) kyaml has a pretty bloated API surface, so I'm wondering if FieldSetter could be enhanced instead.

Yes, I want to set the node for the key, this way I don't break the JSON serialization and other metadata. We might be able to.. but I think the code will get more and more complex over time if we keep adding stuff like this 🤔

Reply:
Looking deeper into that, FieldSetter also removes any null fields, which might not be what we want.
We also have problems with empty objects serialization when we use `FieldSetter`.